From d829a0bb9c25fdbbc4f1a38be55f359b279a5a92 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 28 Jul 2017 17:19:13 +0800
Subject: [PATCH 2310/5242] MLK-16013-31 staging: typec: Set to unattached
 state if souce disconnected

commit  758419568554bb05cdaa10dea8f502b0fc2ab4df from
https://source.codeaurora.org/external/imx/linux-imx.git

In SNK_HARD_RESET_SINK_ON, if the source disconnected, we should set
state to be unattached state intead of startup sink.

Reviewed-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/typec/tcpm.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/tcpm.c b/drivers/usb/typec/tcpm.c
index 7f88944..4c3881f 100644
--- a/drivers/usb/typec/tcpm.c
+++ b/drivers/usb/typec/tcpm.c
@@ -3285,7 +3285,10 @@ static void run_state_machine(struct tcpm_port *port)
 		 * to PE_SNK_Transition_to_default.
 		 */
 		tcpm_set_attached_state(port, true);
-		tcpm_set_state(port, SNK_STARTUP, 0);
+		if (tcpm_port_is_disconnected(port))
+			tcpm_set_state(port, unattached_state(port), 0);
+		else
+			tcpm_set_state(port, SNK_STARTUP, 0);
 		break;
 
 	/* Soft_Reset states */
-- 
1.7.9.5

